### PR TITLE
Improve plos corpus uncorrected proofs checking

### PR DESCRIPTION
**Pull request by eseiver**
There was an error in `plos_corpus` uncorrected_proofs logic that was building the initial proofs list incorrectly (fixed in 6163fdf)

To prevent this sort of error in the future, this PR also:
- fixes `plos_corpus` uncorrected proof functions to make the text_file a parameter that defaults to `uncorrected_proofs_text_list` instead of hard-linking to that file directly (a459610, 
38c529f)
- adds 2 test files to `TESTDATADIR`: one uncorrected proof, one vor update (aa225c5)
- adds unit tests for uncorrected proofs, including `check_for_uncorrected_proofs`(d85adbc)

### CODE REVIEW

File: allofplos/plos_corpus.py [hunk 1]
```diff
@@ -385,7 +385,7 @@ def download_amended_articles(directory=None, tempdir=newarticledir, amended_art
     return amended_updated_article_list
 
 
-def get_uncorrected_proofs_list(directory=None):
+def get_uncorrected_proofs_list(directory=None, text_list=uncorrected_proofs_text_list):
```
Comment (mpacer) [inline allofplos/plos_corpus.py]: I think that you should make the text_list parameter more meaningful… it’s a filepath, right?

File: allofplos/plos_corpus.py [hunk 2]
```diff
@@ -433,7 +433,7 @@ def check_for_uncorrected_proofs(directory=newarticledir, text_list=uncorrected_
     for article_file in articles:
         article = Article.from_filename(article_file)
         article.directory = directory
-        if article.proof == 'uncorrected_proof':
+        if article.proof == 'uncorrected_proof' and article.doi not in uncorrected_proofs_list:
```
Comment (mpacer) [inline allofplos/plos_corpus.py]: This seems more complicated than necessary. Why do you need to check for either? This feels like some layer of abstraction is missing

File: allofplos/tests/test_unittests.py [hunk 3]
```diff
@@ -211,6 +213,17 @@ def test_example_doi2(self):
         self.assertEqual(article.url[:100], "http://journals.plos.org/plosone/article/file?id=10.1371/annotation/3155a3e9-5fbe-435c-a07a-e9a4846e", 'url does not transform correctly for {}'.format(article.doi))
         self.assertEqual(article.word_count, 129, 'word_count does not transform correctly for {}'.format(article.doi))
 
+    def test_proofs(self):
```
Comment (mpacer) [inline allofplos/tests/test_unittests.py]: I really think we need to move the tests into individual files.

**Comment by eseiver** [comment]
OK, by adding an empty temp corpus directory to the tests I got both uncorrected proof functions to pass pytest. Renamed `text_list` parameters to `list_path`. And to avoid clunky duplicate checking in a list, I made the uncorrected proofs functions build and return sets of DOIs instead of lists.

File: allofplos/plos_corpus.py [hunk 4]
```diff
@@ -385,66 +385,67 @@ def download_amended_articles(directory=None, tempdir=newarticledir, amended_art
...
     except FileNotFoundError:
         print("Creating new text list of uncorrected proofs from scratch.")
         article_files = listdir_nohidden(directory)
-        uncorrected_proofs_list = []
+        uncorrected_proofs = set()
         for article_file in tqdm(article_files, disable=None):
             article = Article.from_filename(article_file)
-            if article.proof == 'uncorrected-proof':
-                uncorrected_proofs_list.append(article.doi)
+            article.directory = directory
```
Comment (mpacer) [inline allofplos/plos_corpus.py]: hrm… we should infer the directory from the filename… that may be a separate PR, but we should open an issue about that (cause otherwise this has some weird behaviour).

File: allofplos/plos_corpus.py [hunk 5]
```diff
@@ -385,66 +385,67 @@ def download_amended_articles(directory=None, tempdir=newarticledir, amended_art
...
+        for item in sorted(uncorrected_proofs):
             f.write("%s\n" % item)
-    if uncorrected_proofs_list:
-        print("{} new uncorrected proofs found. {} total in list.".format(new_proofs, len(uncorrected_proofs_list)))
+    if uncorrected_proofs:
+        print("{} new uncorrected proofs found. {} total in set.".format(new_proofs, len(uncorrected_proofs)))
     else:
-        print("No uncorrected proofs found in folder or in existing list.")
-    return uncorrected_proofs_list
+        print("No uncorrected proofs found in folder or in existing file.")
```
Comment (mpacer) [inline allofplos/plos_corpus.py]: instead of saying folder and file, maybe state the directory name and the file name (using `"".format()`)?

File: allofplos/plos_corpus.py [hunk 6]
```diff
@@ -511,8 +512,8 @@ def download_vor_updates(directory=None, tempdir=newarticledir,
         if updated:
             vor_updated_article_list.append(article)
 
-    old_uncorrected_proofs_list = get_uncorrected_proofs_list()
-    new_uncorrected_proofs_list = list(set(old_uncorrected_proofs_list) - set(vor_updated_article_list))
+    old_uncorrected_proofs_list = get_uncorrected_proofs()
```
Comment (mpacer) [inline allofplos/plos_corpus.py]: should probably call it a set now.

File: allofplos/tests/test_unittests.py [hunk 7]
```diff
@@ -211,6 +213,22 @@ def test_example_doi2(self):
...
+        """Tests whether uncorrected proofs and VOR updates are being detected correctly."""
+        article = Article(example_uncorrected_doi, directory=TESTDATADIR)
```
Comment (mpacer) [inline allofplos/tests/test_unittests.py]: you should be able to effect this using `os.environ['PLOS_CORPUS'] = os.path.abspath('test_corpus')` right? 
you shouldn't need to set the directory using `TESTDATADIR`…

File: allofplos/tests/test_unittests.py [hunk 8]
```diff
@@ -211,6 +213,22 @@ def test_example_doi2(self):
...
+        self.assertTrue(article.proof == 'uncorrected_proof')
+        article = Article(example_vor_doi, directory=TESTDATADIR)
+        self.assertTrue(article.proof == 'vor_update')
+        text_file = os.path.join(TESTDIR, 'test.txt')
+        os.makedirs('test_corpus', exist_ok=True)
```
Comment (mpacer) [inline allofplos/tests/test_unittests.py]: why don't you just point it at the testdata directory? how does `TESTDIR` differ from `TESTDATADIR`

File: allofplos/plos_corpus.py (hunk 4 above)
Comment (eseiver) [reply allofplos/plos_corpus.py]: indeed; see #52

Comment (mpacer) [reply allofplos/plos_corpus.py]: lol! did not realise there was such an old issue on this!

File: allofplos/plos_corpus.py [hunk 9]
```diff
@@ -569,15 +572,15 @@ def remote_proofs_direct_check(tempdir=newarticledir, article_list=None, plos_ne
     return proofs_download_list
 
 
-def download_check_and_move(article_list, text_list, tempdir, destination,
+def download_check_and_move(article_list, list_path, tempdir, destination,
```
Comment (mpacer) [inline allofplos/plos_corpus.py]: could `list_path` instead be `proof_filepath`? i'm just realising that list_path is way too generic to be clear in the signature of this function

File: allofplos/plos_corpus.py [hunk 10]
```diff
@@ -385,66 +385,69 @@ def download_amended_articles(directory=None, tempdir=newarticledir, amended_art
...
-def check_for_uncorrected_proofs(directory=newarticledir, text_list=uncorrected_proofs_text_list):
+def check_for_uncorrected_proofs(directory=newarticledir, list_path=uncorrected_proofs_text_list):
     """
     For a list of articles, check whether they are the 'uncorrected proof' type
     One of the checks on newly downloaded articles.
-    :param text_list: List of DOIs
+    :param list_path: List of DOIs
     :param directory: Directory containing the article files
-    :return: all articles that are uncorrected proofs, including from main article directory
+    :return: set of all articles that are uncorrected proofs, including from main article directory
```
Comment (mpacer) [inline allofplos/plos_corpus.py]: isn't it a set of all dois?

File: allofplos/plos_corpus.py (hunk 9 above)
Comment (mpacer) [reply allofplos/plos_corpus.py]: wait… why does list_path need to be here at all? it doesn't look like its used at all

**Comment by mpacer** [comment]
Ok… I think this is good since we want to remove this code eventually, mreging
